### ISSUE TITLE
registerWith gives NPE when using Flutter's background execution on Android

### ISSUE THREAD

**Issue by ryanheise** [dlutton/flutter_tts#13]
## 🐛 Bug Report

Flutter's new background execution feature (described here: https://medium.com/flutter-io/executing-dart-in-the-background-with-flutter-plugins-and-geofencing-2b3e40a1a124) allows plugins to be registered in a background context (e.g. a Service). The problem is that flutter_tts assumes that the context for plugin registration is an activity with this line of code:

`		channel.setMethodCallHandler(new FlutterTtsPlugin(registrar.activity(), channel));`

`registrar.activity()` may now return null, and this leads to a NPE:

```
E/AndroidRuntime(12425): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.Context android.app.Activity.getApplicationContext()' on a null object reference
E/AndroidRuntime(12425): 	at com.eyedeadevelopers.fluttertts.FlutterTtsPlugin.<init>(FlutterTtsPlugin.java:32)
E/AndroidRuntime(12425): 	at com.eyedeadevelopers.fluttertts.FlutterTtsPlugin.registerWith(FlutterTtsPlugin.java:82)
E/AndroidRuntime(12425): 	at io.flutter.plugins.GeneratedPluginRegistrant.registerWith(GeneratedPluginRegistrant.java:23)
```

The solution is to change the above line of code to this:

`		channel.setMethodCallHandler(new FlutterTtsPlugin(registrar.activeContext() , channel));`

(and change the declared type of the constructor's first parameter accordingly from Activity to the more general Context).

I am unaware of whether there is a similar bug on the iOS side.

### Expected behavior

No NPE.

### Reproduction steps

There are 3 sample projects from Google that I am aware of that demonstrate background execution:

https://github.com/bkonyi/FlutterGeofencing
https://github.com/flutter/plugins/tree/master/packages/android_alarm_manager
https://github.com/flutter/plugins/tree/master/packages/location_background

The first is for Android/iOS. The second is Android only. The third is iOS only.

To reproduce the problem, just add the flutter_tts plugin as a dependency to one of the first two projects and run the app (you may need to use the appropriate feature of the app that triggers background execution).

### Configuration

**Version:** 0.1.1

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android

**Comment by ryanheise** [comment]
Hi Daniel,

I've just released the project in which I encountered this bug as an open source flutter plugin. See https://pub.dartlang.org/packages/audio_service . This might provide an easier way for you to reproduce the bug:

1. git clone https://github.com/ryanheise/audio_service
2. add flutter_tts to example/pubspec.yaml
3. open example/lib/main.dart and change the line `await soundpool.play(clickSoundId);` to something like `await tts.speak('hello');` where `tts` is an instance of your `FlutterTts` class.
4. run the example, press play and see the `NullPointerException`.

The fix is fairly safe (I've tried it myself by modifying your code). i.e. to simply change `registrar.activity()` to `registrar.activeContext()`.  I'd love to use your flutter_tts in the audio_service example app if this bug can be fixed.

**Comment by dlutton** [comment]
Thank you for the example, I'll attempt the change and test it out.

**Comment by ryanheise** [comment]
I've implemented this in pull request #15

**Comment by dlutton** [comment]
Tested and it looks good.  I have merged the PR.  Thank you!
